### PR TITLE
Deniability middleware, some util packages

### PR DESCRIPTION
**Pull request by mrd0ll4r**
Contributes to #145

### CODE REVIEW

File: middleware/deniability/README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+## Deniability Middleware
+
+This package provides the announce middleware `deniability` which inserts ghost peers into announce responses to achieve deniability.
```
Comment (jzelinskie) [inline middleware/deniability/README.md]: to achieve plausible deniability

File: middleware/deniability/config.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+	"github.com/chihaya/chihaya"
+)
+
+// Config represents the configuration for the deniability middleware.
+type Config struct {
+	// ModifyResponseProbability is the probability by which a response will
+	// be augmented with random peers.
+	ModifyResponseProbability float32 `yaml:"modify_response_probability"`
+	// MaxRandomPeers is the amount of peers that will be added at most.
```
Comment (jzelinskie) [inline middleware/deniability/config.go]: newlines between fields

File: middleware/deniability/config_test.go [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+	maxNewPeers       string
+	prefix            string
+	minPort           string
+	maxPort           string
+	err               bool
+	expected          Config
+}
+
+var (
+	configTemplate = `name: foo
```
Comment (jzelinskie) [inline middleware/deniability/config_test.go]: just break this first newline

```go
var configTemplate = `
name: foo
config:
modify_response_probability: %s
max_random_peers: %s
prefix: %s
min_port: %s
max_port: %s
`
```

File: middleware/deniability/config_test.go [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+		mwconfig chihaya.MiddlewareConfig
+		config   string
+	)
+
+	cfg, err := newConfig(mwconfig)
+	assert.Nil(t, err)
+	assert.NotNil(t, cfg)
+
+	for _, test := range configData {
+		config = fmt.Sprintf(configTemplate, test.modifyProbability, test.maxNewPeers, test.prefix, test.minPort, test.maxPort)
```
Comment (jzelinskie) [inline middleware/deniability/config_test.go]: use `:=` to get rid of the var block at the top

File: middleware/deniability/deniability_test.go [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+	"github.com/chihaya/chihaya"
+)
+
+type constructorTestData struct {
+	cfg   Config
+	error bool
+}
+
+var constructorData = []constructorTestData{
+	{Config{1.0, 10, "abc", 1024, 1025}, false},
```
Comment (jzelinskie) [inline middleware/deniability/deniability_test.go]: Do you need `Config` here?

Comment (mrd0ll4r) [inline middleware/deniability/deniability_test.go]: yup, "missing type in composite literal"

File: pkg/randstring/generate_test.go [hunk 6]
```diff
@@ -0,0 +1,26 @@
+package randstring
```
Comment (jzelinskie) [inline pkg/randstring/generate_test.go]: copyright header is missing

File: pkg/randpeer/rand_peer_test.go [hunk 7]
```diff
@@ -0,0 +1,39 @@
+package randpeer
```
Comment (jzelinskie) [inline pkg/randpeer/rand_peer_test.go]: copyright header

**Comment by jzelinskie** [comment]
Can we consolidate the randstring and randpeer into one random package?

File: pkg/random/peer.go [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+// v6 indicates whether an IPv6 address should be generated.
+// Regardless of the length of the generated IP address, its bytes will have
+// values in [1,254].
+//
+// minPort and maxPort describe the range for the randomly generated port, where
+// minPort <= port < maxPort.
+// minPort and maxPort will be checked and altered so that
+// 1 <= minPort <= maxPort <= 65536.
+// If minPort == maxPort, port will be set to minPort.
+//
```
Comment (jzelinskie) [inline pkg/random/peer.go]: get rid of this last empty comment line

**Comment by jzelinskie** [comment]
If you remove that last little line, this LGTM

**Comment by jzelinskie** [comment]
LGTM. Thanks again!

**Pull request merged**